src/Plugin/formatter/FormatterHalJson.php

```diff
@@ -121,14 +121,14 @@ protected function addHateoas(array &$data) {
     );
 
     $input = $request->getParsedInput();
-    $data_provider = $resource->getDataProvider();
     $page = !empty($input['page']) ? $input['page'] : 1;
 
+    $query = $input;
     if ($page > 1) {
-      $input['page'] = $page - 1;
+      $query['page'] = $page - 1;
       $data['_links']['previous'] = array(
         'title' => 'Previous',
-        'href' => $resource->getUrl(),
+        'href' => $resource->versionedUrl('', array('query' => $query), TRUE),
```

Comment:
No, just keep it consistent with [FormatterJson#L168](https://github.com/RESTful-Drupal/restful/blob/7.x-2.x/src/Plugin/formatter/FormatterJson.php#L168).